### PR TITLE
refactor: migrate 'icons-*' to TS

### PR DESCRIPTION
**Pull request by ilhan007**
Migrate the source code base of "icons-*" packages to Typescript
Mapped "**dist**" resources to return result of type "**CollectionData**"
- */generated/assets/SAP-icons-business-suite.json
- */generated/assets/SAP-icons-TNT.json
- *generated/assets/v4/SAP-icons.json
- *generated/assets/v5/SAP-icons.json

Added option **"module": "ESNext"** to tsconfig to enable usage of dynamic imports

### CODE REVIEW

**Review by pskelin — APPROVED** [review]
looks good. typescript becomes mandatory for icons packages, but leave it as it is. If anyone else makes a package, they can always modify the `package-scripts.js`

**Review by vladitasev — DISMISSED** [review]
The `tsc` command must be enabled only for our libraries, at least in theory, but I don't know of any stakeholder who created an icon library, so probably good to merge like this.

File: packages/icons-business-suite/package.json [hunk 1]
```diff
@@ -26,7 +26,8 @@
     "directory": "packages/icons-business-suite"
   },
   "dependencies": {
-    "@ui5/webcomponents-base": "1.9.1"
+    "@ui5/webcomponents-base": "1.9.1",
+    "typescript": "^4.9.3"
```
Comment (MarcusNotheis) [inline packages/icons-business-suite/package.json:30]: Why do we need typescript as a dependency? In my opinion it should be a dev dependency. Same applies for the other packages.

Comment (ilhan007) [reply packages/icons-business-suite/package.json:30]: We will change that, thank you @MarcusNotheis

**Pull request merged**